### DISCUSSION TITLE
Support exporting tables

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#2268] [Feature Ideas]
Mathesar's UI should support exporting tables (and explorations) in common formats.

---
*If you'd like us to build this feature, please comment below with more details about how you plan to use this functionality.*

**Comment by jeromeperez** [comment]
It should be very interesting for my user to have possibility to export data and exploration to show their work to other people without access to database

**Reply by kgodey** [reply]
@jeromeperez Thanks for letting us know! What kind of formats would you like to export the data to?

Also, we already have a "share" feature for tables and explorations, this creates a public read-only URL which is visible to anyone, no account needed. Is this useful for your use case?

**Reply by jeromeperez** [reply]
in my case a csv and/or tsv format are better format.
The share feature is not interesting in my case because searchers have to work on data i send with R, python or other software

**Reply by kgodey** [reply]
Thank you @jeromeperez!

**Reply by jeromeperez** [reply]
This feature of exporting data or exploration in a tsv file is very important for my searchers, i hope you will add it

**Reply by kgodey** [reply]
This is high on our priority list, but we'll need to finish our current backend and permissions refactor before we will be able to work on this.

**Reply by jeromeperez** [reply]
i can wait ;-)

**Comment by HoldYourWaffle** [comment]
Our main use-case for this would be an "escape hatch" that allows users to fall back to the treacherous realm of barebones spreadsheets when Mathesar for whatever reason is not sufficient. This would mainly be an XLSX export, preferably with as much formatting preserved as possible.
Similarly, CSV exports can be crucial for forwarding reports to other systems.

**Reply by kgodey** [reply]
Thanks @HoldYourWaffle! We're actually currently working on this, we're starting out with CSV exports.

**Reply by jeromeperez** [reply]
what a good news ;-)

**Comment by loleg** [comment]
What would be great here is if you'd also export the Metadata of the table (provenance attributes, schema, etc.) using a lightweight format like [Frictionless Data](https://github.com/frictionlessdata/).
